Modelica/Mechanics/Rotational/Components/Clutch2.mo

```diff
@@ -0,0 +1,171 @@
+within Modelica.Mechanics.Rotational.Components;
+model Clutch2 "Clutch based on Coulomb friction (with interpolation based on CombiTable1Ds)"
+  extends Modelica.Mechanics.Rotational.Icons.Clutch;
+  extends Modelica.Mechanics.Rotational.Interfaces.PartialCompliantWithRelativeStates;
+
+  parameter Real mu_pos[:, 2] = [0, 0.5]
+    "Positive sliding friction coefficient [-] as function of w_rel [rad/s] (w_rel>=0)";
+  parameter Real peak(final min=1) = 1
+    "Peak for maximum value of mu at w==0 (mu0_max = peak*mu_pos[1,2])";
+  parameter Real cgeo(final min=0) = 1
+    "Geometry constant containing friction distribution assumption";
+  parameter SI.Force fn_max(final min=0, start=1) "Maximum normal force";
+
+  extends Rotational.Interfaces.PartialFriction;
+  extends Modelica.Thermal.HeatTransfer.Interfaces.PartialElementaryConditionalHeatPortWithoutT;
+
+  Real mu "Friction coefficient";
+  SI.Force fn "Normal force (fn=fn_max*f_normalized)";
+  Modelica.Blocks.Interfaces.RealInput f_normalized
+    "Normalized force signal 0..1 (normal force = fn_max*f_normalized; clutch is engaged if > 0)"
+    annotation (Placement(transformation(origin={0,110}, extent={{20,-20},{-20,20}}, rotation=90)));
+
+protected
+  final parameter Real mu0 = Modelica.Math.Vectors.interpolate(mu_pos[:,1], mu_pos[:,2], 0, 1)
+    "Friction coefficient for w=0 and forward sliding" annotation(Evaluate = true);
+
+  Real table_signs[2]
+    "Signs for sliding friction coefficient table interpolation: [sign for w_rel, sign for mu]";
+  Modelica.Blocks.Tables.CombiTable1Ds table(
+    table=mu_pos) "Sliding friction coefficient table";
+
+equation
+  // Relative quantities
+  w_relfric = w_rel;
+  a_relfric = a_rel;
+
+  // Normal force and friction torque for w_rel=0
+  fn = fn_max*f_normalized;
+  free = fn <= 0;
+  tau0 = mu0*cgeo*fn;
+  tau0_max = peak*tau0;
+
+  // Friction torque
+  table_signs = if startForward then
+      {1,1}
+    elseif startBackward then
+      {1,-1}
+    elseif pre(mode) == Forward then
+      {1,1}
+    else
+      {-1,-1};
```

Comment:
Possible to change to (for even better readability):
```
  table_signs =
    if     startForward         then { 1, 1}
    elseif startBackward        then { 1,-1}
    elseif pre(mode) == Forward then { 1, 1}
    else                             {-1,-1};
```

Reply:
Sure, go ahead.